pandas/core/indexes/multi.py

```diff
@@ -2709,7 +2709,10 @@ def _partial_tup_index(self, tup: tuple, side: Literal["left", "right"] = "left"
                     # non-comparable level, e.g. test_groupby_example
                     raise TypeError(f"Level type mismatch: {lab}")
                 if side == "right" and loc >= 0:
-                    loc -= 1
+                    # error: Incompatible types in assignment (expression has type
+                    # "Union[int, Any]", variable has type "Union[ndarray[Any,
+                    # dtype[signedinteger[Any]]], signedinteger[Any]]")
+                    loc -= 1  # type: ignore[assignment]
```

Comment:
Might be a little rusty on my python typing but I think it just infers the value from the first appearing of a variable within a function. `loc` is up higher in the func from `algos.searchsorted` - would that ever return a scalar? If not we might just want to be explicit that `loc` can be an `int` as well

Reply:
algos.searchsorted returns `npt.NDArray[np.intp] | np.intp`.  A couple lines of we should have ruled out the ndarray case.  so we have np.intp and its complaining that it isnt the same as int.